valkeyotel/metrics.go

```diff
@@ -84,6 +85,10 @@ func NewClient(clientOption valkey.ClientOption, opts ...Option) (valkey.Client,
 		return nil, err
 	}
 
+	if len(clientOption.InitAddress) == 1 {
+		oclient.tAttrs = append(oclient.tAttrs, serverAttrs(clientOption.InitAddress[0]))
+	}
```

Comment:
This obviously won't cover everything, but it should cover some very common cases (e.g. AWS ElastiCache serverless clusters) and leave room for others with more complex topologies to contribute further detections in the future.